### ISSUE TITLE
http

### ISSUE THREAD

**Issue by Bossfrosty** [jailsonsb2/RadioplayerAPI#2]
I apologize for posting this question in the Issues section, as I'm not sure where else to ask. I'm having trouble streaming a radio station with an HTTP URL. While radio stations with HTTPS URLs work fine, those with only HTTP URLs won't play when accessed through a web server using HTTPS. Could someone please help me understand why this is the case?

**Comment by jailsonsb2** [comment]
You're absolutely right, this is a common issue. The reason for this is a security measure called "Mixed Content Blocking" implemented by modern web browsers. Here's the breakdown:

* **HTTPS (Hyper Text Transfer Protocol Secure):** This is a secure protocol used to transmit data over the internet. It encrypts the data, protecting it from unauthorized access.
* **HTTP (Hyper Text Transfer Protocol):** This is the standard protocol for transferring data but it doesn't encrypt data, making it less secure.
* **Mixed Content:** This occurs when a secure HTTPS website tries to load resources (like images, videos, or audio streams) from an insecure HTTP source.

Browsers block mixed content to protect users from potential security risks. When an HTTPS website loads HTTP content, it creates a vulnerability that attackers could exploit to insert malicious content or steal sensitive information.

**How to Solve This Issue:**

There are a few ways to address this issue:

1. **Contact the Radio Station:** The best solution is to ask the radio station to provide an HTTPS stream. Many stations are making the switch to HTTPS to improve security and compatibility.
2. **Use a Proxy:** A proxy server can act as an intermediary between your website and the radio station. It can fetch the HTTP stream and deliver it to your website over HTTPS.
3. **Use a Radio Player Plugin:** Some plugins for content management systems (like WordPress) offer features to proxy HTTP streams, making them accessible on HTTPS websites.

**Example Using a Radio Player Plugin (WordPress):**

1. **Install and Activate:** Find a radio player plugin that supports HTTP stream proxying (like "Radio Player" or "Radio Station").
2. **Enable Proxy:** In the plugin settings, enable the proxy feature.
3. **Add Stream URL:** Enter the HTTP URL of the radio station in the plugin.
4. **Embed Player:** Embed the radio player on your website. The plugin will handle the proxying, allowing the HTTP stream to play on your HTTPS website.

**Important Considerations:**

* **Security:** Using a proxy or plugin can introduce additional security considerations. Ensure that you choose reputable solutions and keep them updated.
* **Performance:** Proxying can slightly increase the load on your server. Consider the impact on performance if you have a high-traffic website.

I hope this explanation helps! If you have any more questions, feel free to ask.
